src/VisualStudio/Core/Def/Implementation/TaskList/CommentTaskTokenSerializer.cs

```diff
@@ -77,7 +77,7 @@ private void OnPropertyChanged(object sender, PropertyChangedEventArgs e)
             _lastCommentTokenCache = commentString;
 
             // let people to know that comment string has changed
-            _optionService.SetOptions(optionSet.WithChangedOption(TodoCommentOptions.TokenList, _lastCommentTokenCache));
+            _workspace.SetOptions(optionSet.WithChangedOption(TodoCommentOptions.TokenList, _lastCommentTokenCache));
```

Comment:
>_workspace [](start = 12, length = 10)

Why use workspace instead of IOptionsService? Do we not want this to update the option for all workspaces that share the option service?

Reply:
Workspace.SetOptions will end up setting the options via the options service, which will in turn update the current solution of each of the registered workspaces to have the new options. I think each component that is attached to a specific workspace should only set options for its own workspace. Eventually, I hope we can make the option service option setter to be accessible only from Workspace.